Tools/cases_generator/generate_cases.py

```diff
@@ -734,6 +734,57 @@ def stack_analysis(
         ]
         return stack, -lowest
 
+    def get_stack_effect_info(self, thing):
```

Comment:
Can you type-annotate the non-self arguments and return types (even if `-> None`) of the functions you define? This file is currently clean (in pyright, at least) and I'd like to keep it that way.